examples/tests/bare_bones/main.c

```diff
@@ -0,0 +1,29 @@
+// Do nothing. yield() immediately.
+
+// Define the `_start_` signature.
+void _start(void* app_start, void* mem_start, void* memory_len, void* app_heap_break);
+
+void _start(void* app_start __attribute__((unused)),
+            void* mem_start __attribute__((unused)),
+            void* memory_len __attribute__((unused)),
+            void* app_heap_break __attribute__((unused))) {
+  // Yield.
+#if defined(__thumb__)
+  __asm__ volatile (
+      "svc 0       \n"
+      :
+      :
+      : "memory", "r0", "r1", "r2", "r3", "r12", "lr"
+      );
+
+#elif defined(__riscv)
+  __asm__ volatile (
+      "li    a0, 0\n"
```

Comment:
This doesn't look right. AFAIK the system call type is passed in `a4` on RISC-V, with `a0` to `a3` being the arguments (where for yield we're probably going to want to load a well-defined value into `a0` anyways).
```suggestion
      "li    a0, 1\n" // yield-wait
      "li    a4, 0\n" // yield system call class
```

Reply:
Ohhhh, ok I'm not crazy. a0 was the correct register when I wrote this app.